### ISSUE TITLE
Unexpected default placement of deleted item when sibling can't be found

### ISSUE THREAD

**Issue by noah-potter** [react-spring/react-spring#737]
## 🐛 Bug Report

Not sure if this is a bug or intentional, but it feels like a bug to me.
I'm observing this result:
```js
transitions = useTransition(1, null, { ... })
// transitions = [{ item:1, ...}]

// and later this is run:
transitions  = useTransition(2, null, { ... })
// transitions = [{ item:2, ...}, { item:1, ...}]
// my ideal would be: transitions = [{ item:1, ...}, { item:2, ...}]
```
The resulting transitions array while transitioning puts the CREATED item first, and I feel it should come last, the idea being to preserve the order of the array as closely as possible. I've tracked this down to the line in the "sanity check" here in `reconcileDeleted`:
https://github.com/react-spring/react-spring/blob/512f1e849d6d94729f0fce929e531e1a3249a318/packages/core/src/helpers.ts#L120
It can't find a sibling index because there were no siblings during the first `useTransition`, and so the sanity check is triggered and puts the deleted value last using `.push`

I'm thinking it's as easy as using `.unshift` instead but I'm not familiar with this project enough to know if that change is significant, so I wanted to bring this up as an issue first. It may not be that simple since this seems to be an edge case and using `.unshift` would be convenient, but may be the wrong thing to do. 

I'm working on getting the repo running to create a (hopefully) failing test to confirm this, and to see if any other tests break if I change it to `.unshift`, but I'm having trouble getting it to run on windows, looks like it's expecting unix paths? I'll have to setup my laptop if that's the case, so in the meantime I wanted to throw this out there and see if this makes sense to anyone without examples/tests.

Oh and the reason I need the order of items preserved as much as possible is because I'm rendering overlapping elements and I need the newest items to be rendered last so that they're rendered on top of older elements, but as it stands in that example, the oldest (deleted) item is rendered last since it's last in the `transitions` array.

## Environment

- `react-spring` v9.0.0-beta.30
- `react` v16.8.6

**Comment by aleclarson** [comment]
> but I'm having trouble getting it to run on windows, looks like it's expecting unix paths?

We should fix this. Where are you having issues on Windows? In the tests?

> Oh and the reason I need the order of items preserved as much as possible is because I'm rendering overlapping elements and I need the newest items to be rendered last so that they're rendered on top of older elements, but as it stands in that example, the oldest (deleted) item is rendered last since it's last in the `transitions` array.

I wonder if similar use cases exist where multiple "items" are used. If so, we probably don't want a special case for single "items". I didn't write `useTransition`, so I'm not sure if there's a good reason for putting deleted "items" last in the returned array. @drcmda Could you enlighten us here?

**Comment by noah-potter** [comment]
Yeah, running the tests on windows doesn't work. I believe the issue was in `jest.config.js`. On unix, it's passing paths like `/a/b/c` to recrawl, which puts that into a regex as-in, but on windows it's passing `\a\b\c` which recrawl puts into the regex, but the regex reads the backslashes as escapes and so breaks.

>I wonder if similar use cases exist where multiple "items" are used. If so, we probably don't want a special case for single "items". Also, I didn't write useTransition, so I'm not sure if there's a good reason for putting deleted "items" last in the returned array. @drcmda Could you enlighten us here?

Once I got a test in place, I realized my initial idea of using `unshift` would be a bit hacky and it broke other tests anyways, so I went with explicitly handling the case that there is no `left` or `right` to an item, short-circuiting the entire logic that tries to find a sibling. See the PR attached to this issue, I think it turned out to be pretty simple and covers an edge case that was missed.
